core/trino-spi/src/main/java/io/trino/spi/block/AbstractRowBlock.java

```diff
@@ -194,7 +214,7 @@ public Block getSingleValueBlock(int position)
         for (int i = 0; i < numFields; i++) {
             newBlocks[i] = getRawFieldBlocks()[i].copyRegion(startFieldBlockOffset, fieldBlockLength);
         }
-        boolean[] newRowIsNull = new boolean[] {isNull(position)};
+        boolean[] newRowIsNull = isNull(position) ? new boolean[] {true} : null;
```

Comment:
`new boolean[] { true }` could be a constant.

Reply:
Could be, but that constant would probably be better shared between block types (since `getSingleValueBlock(position)` will always be either `null` or `new boolean[]{true}`), which implies a bigger refactor than the scope of this PR.